activerecord/test/cases/associations/nested_through_associations_test.rb

```diff
@@ -538,7 +538,7 @@ def test_nested_has_many_through_with_conditions_on_source_associations
   end
 
   def test_nested_has_many_through_with_conditions_on_source_associations_preload
-    author = assert_queries(4) { Author.includes(:misc_post_first_blue_tags_2).third }
+    author = assert_queries(2) { Author.includes(:misc_post_first_blue_tags_2).third }
```

Comment:
Before the change, `assert_queries(4)` (author -> posts -> taggings -> tags) was caused due to lost the source reflection scopes.

https://github.com/rails/rails/blob/8538dfdc084555673d18cfc3479ebef09f325c9c/activerecord/test/models/author.rb#L151-L155

https://github.com/rails/rails/blob/8538dfdc084555673d18cfc3479ebef09f325c9c/activerecord/test/models/post.rb#L135-L137

`Post.has_many :first_blue_tags_2` references the through `taggings` table, so the association should not be load one by one queries theoretically.